opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/websocket/WebsocketAutoConfiguration.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.websocket;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.support.ExecutorSubscribableChannel;
+import io.opentracing.Tracer;
+import io.opentracing.tag.Tags;
+
+@Configuration
```

Comment:
It should be conditional on tracer bean

Reply:
When I made it conditional on the tracer bean, and used it with an example spring-boot app, it didn't report any traces.

The problem is that if nothing autowires the Tracer, then its injection is not triggered. So if all of the instrumentation auto-configs are protected by ConditionalOnBean(Tracer), then what is going to cause the Tracer to be resolved?

Reply:
When there is no tracer it should always create `NoopTracer` so all other auto configs should kick in.

Reply:
In the example I was testing, I had Jaeger configured. The only change between it working and not was commenting out the ConditionalOnBean. From a quick google its not clear whether @Bean 's are instantiated lazily - but the behaviour seemed to indicate that if there was no (unconditional) `@Autowire Tracer` then it didn't try to get the tracer.

But will add the `ConditionalOnBean` for now and can be investigated later if still is an issue.

Reply:
Although actually from your statement about `NoopTrace`, this would imply that the `ConditionalOnBean` would be redundant, as there would always be a tracer - even if just a `NoopTracer`?

Reply:
> Although actually from your statement about NoopTracer, this would imply that the ConditionalOnBean would be redundant, as there would always be a tracer - even if just a NoopTracer?

You are right. Originally (even in java-spring-web) auto configs didn't depend on Tracer bean. I assumed that there is always an instance of the tracer. The change was implemented by one of the external contributors. If you look at sleuth there is also an assumption that tracer will be there and it is conditional on tracer bean. 

Let's go with the same approach like it is in other auto-configs in this repo and keep it consistent. We can remove it later.